modules/idaithalam/src/main/java/io/virtualan/idaithalam/core/generator/FeatureFileGenerator.java

```diff
@@ -145,22 +145,24 @@ private static void addCustomApiHeader(ApiExecutorParam apiExecutorParam, List<I
             for (Map<String, Object> map : apiHeaderList) {
                 for (String key : map.keySet()) {
                     AvailableParam newAvailableParam = new AvailableParam(key, map.get(key).toString(), "HEADER_PARAM");
-                    if (!item.getAvailableParams().contains(newAvailableParam)) {
+                    if (!item.getAvailableParams().contains(newAvailableParam)) { 
                         item.getAvailableParams().add(newAvailableParam);
                         item.getHeaderParams().add(newAvailableParam);
-                    } else if (overwrite) { //if it exists already, overwrite it
+                        item.setHasHeaderParams(true);
+                    } else { //if it exists already, add value as a value list or overwrite it depending on configuration.
                         for (AvailableParam availableParam1 : item.getAvailableParams()) {
                             if (availableParam1.getKey().equals(newAvailableParam.getKey())) {
-                                availableParam1.setKey(newAvailableParam.getKey());
-                                availableParam1.setValue(newAvailableParam.getValue());
-                                availableParam1.setParameterType(newAvailableParam.getParameterType());
-                                LOGGER.warning("Due to issue #121 API header " + key + " is overwritten with value from configuration. To avoid this behavior add 'overwrite: false' (default: true) to the 'apiHeader' section.");
+                                if (overwrite) {
```

Comment:
move this code to separate method

Reply:
You will get a new method with 4 - 5 input parameter for that task.
I would not make a new method with that.